src/main/java/com/wavefront/sdk/grpc/reporter/WavefrontGrpcReporter.java

```diff
@@ -0,0 +1,185 @@
+  public void registerGauge(MetricName metricName, AtomicInteger atomicInteger) {
+    wfReporter.newGauge(metricName, () -> (double) atomicInteger.get());
+  }
+
+  @Override
+  public void start() {
+    wfReporter.start(reportingIntervalSeconds, TimeUnit.SECONDS);
+  }
+
+  @Override
+  public void stop() {
+    if (serverHeartbeat != null) {
+      serverHeartbeat.close();
+    }
+    if (clientHeartbeat != null) {
+      clientHeartbeat.close();
+    }
+  }
+
+  public synchronized void registerServerHeartBeat() {
+    if (serverHeartbeat != null) {
+      serverHeartbeat = new HeartbeaterService(wavefrontMetricSender, applicationTags,
+          GRPC_SERVER_COMPONENT);
+    }
+  }
+
+  public synchronized void registerClientHeartbeat() {
+    if (clientHeartbeat != null) {
+      clientHeartbeat = new HeartbeaterService(wavefrontMetricSender, applicationTags,
+          GRPC_CLIENT_COMPONENT);
+    }
+  }
+
+  public static class Builder {
+    // Required parameters
+    private final ApplicationTags applicationTags;
+    private final String prefix = "grpc";
+
+    // Optional parameters
+    private int reportingIntervalSeconds = 60;
+
+    @Nullable
+    private String source;
+
+    /**
+     * Builder to build WavefrontGrpcReporter.
+     *
+     * @param applicationTags  metadata about your application that you want to be propagated as
+     *                         tags when metrics/histograms are sent to Wavefront.
+     */
+    public Builder(ApplicationTags applicationTags) {
+      this.applicationTags = applicationTags;
+    }
+
+    /**
+     * Set reporting interval i.e. how often you want to report the metrics/histograms to
+     * Wavefront.
+     *
+     * @param reportingIntervalSeconds reporting interval in seconds.
+     * @return {@code this}.
+     */
+    public Builder reportingIntervalSeconds(int reportingIntervalSeconds) {
+      this.reportingIntervalSeconds = reportingIntervalSeconds;
+      return this;
+    }
+
+    /**
+     * Set the source tag for your metric and histograms.
+     *
+     * @param source Name of the source/host where your application is running.
+     * @return {@code this}.
+     */
+    public Builder withSource(String source) {
+      this.source = source;
+      return this;
+    }
+
+    /**
+     * Build a WavefrontGrpcReporter.
+     *
+     * @param wavefrontSender send data to Wavefront via proxy or direct ingestion.
+     * @return An instance of {@link WavefrontGrpcReporter}.
+     */
+    public WavefrontGrpcReporter build(WavefrontSender wavefrontSender) {
+      if (source == null) {
+        try {
+          source = InetAddress.getLocalHost().getHostName();
+        } catch (UnknownHostException e) {
+          // Should never happen
+          source = "unknown";
+        }
+      }
+      // application tag is reported with every metric
+      Map<String, String> pointTags = new HashMap<>();
+      pointTags.put("application", applicationTags.getApplication());
```

Comment:
You created the final string in tier1 SDK right?

Reply:
yes, fixed.